mono/mini/aot-compiler.c

```diff
@@ -4721,7 +4721,7 @@ add_generic_class_with_depth (MonoAotCompile *acfg, MonoClass *klass, int depth,
 	if (klass->image == mono_defaults.corlib && !strcmp (klass->name_space, "System.Collections.Generic") &&
 		(!strcmp(klass->name, "ICollection`1") || !strcmp (klass->name, "IEnumerable`1") || !strcmp (klass->name, "IList`1") || !strcmp (klass->name, "IEnumerator`1") || !strcmp (klass->name, "IReadOnlyList`1"))) {
 		MonoClass *tclass = mono_class_from_mono_type (mono_class_get_generic_class (klass)->context.class_inst->type_argv [0]);
-		MonoClass *array_class = mono_bounded_array_class_get (tclass, 1, FALSE);
+		MonoClass *array_class = mono_bounded_array_class_create (tclass, 1, FALSE);
```

Comment:
Someone will object to name lengthening?

Reply:
But it is probably good -- get tends to retrieve available state, create tends to create new/additional state.

Reply:
The main difference is that `mono_bounded_array_get` is `MONO_API`, so fixed in stone.  `mono_bounded_array_class_create` can have its semantics change: for example to be relaxed to create array classes that are not fully initialized.